ext/standard/tests/streams/stream_cast_loses_data.phpt

```diff
@@ -0,0 +1,162 @@
+--TEST--
+Casting a stream can lose data and needs to emit a warning
+--FILE--
+<?php
+$stream = popen('for i in $(seq 1 2000); do echo $i; done', 'r');
+sleep(2);
+
+// fills the read buffer with up to 8192 bytes
+fgets($stream);
+
+// cast $stream and read fd until eof. Print each line that was read, prefixed with "proc open stdin:"
+proc_open(
+    'sed "s/^/proc open stdin:/"',
+    [
+        0 => $stream,
+    ],
+    $pipes,
+);
+?>
+--EXPECTF--
+Warning: proc_open(): 8190 bytes of buffered data lost during stream conversion! in %s on line %d
+proc open stdin:
+proc open stdin:1861
+proc open stdin:1862
```

Comment:
Thank you for including this test.

We could make it more robust and also remove the sleep(2) like this: https://gist.github.com/arnaud-lb/043b8649d6daf0c6433791ff7212b710

Reply:
Done :)